### PR TITLE
TG-45, TG-47 QR codes for enrollment

### PR DESCRIPTION
**Pull request by gajus123**

### CODE REVIEW

File: apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/api/EnrollmentController.java [hunk 1]
```diff
@@ -46,4 +46,14 @@
...
     }
+
+    @ApiOperation(value = "Get enrollment QR code", nickname = "get enrollment QR code", notes = "",
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "If valid credentials were provided", response = Iterable.class),
+            @ApiResponse(code = 400, message = "If invalid data was provided")})
+    @GetMapping("{id}/qr-code")
+    ResponseEntity<byte[]> getQRCode(@PathVariable Long id) {
+        return ResponseEntity.ok(enrollmentService.generateQRCode(id));
```
Comment (Delebrith) [inline apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/api/EnrollmentController.java:57]: I think maybe a content type header should be added to the response, so the browser can interpret it as an image. Does swagger show a weird text in response or an image? With content type it shoud be an image that can be downloaded.

File: apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/qrcode/exception/QRCodeGenException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package edu.pw.apsienrollment.qrcode.exception;
+
+import edu.pw.apsienrollment.common.exception.ApsiException;
+import edu.pw.apsienrollment.common.exception.ExceptionCode;
+
+public class QRCodeGenException extends ApsiException {
+    public QRCodeGenException() {
+        super(ExceptionCode.QR_CODE_GEN_FAILED, "Error during QR code generation", null);
```
Comment (Delebrith) [inline apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/qrcode/exception/QRCodeGenException.java:8]: Please, add translation of this error code on FE

Comment (gajus123) [reply apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/qrcode/exception/QRCodeGenException.java:8]: Done

File: apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/api/EnrollmentController.java (hunk 1 above)
Comment (gajus123) [reply apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/api/EnrollmentController.java:58]: Done

File: apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/EnrollmentServiceImpl.java [hunk 3]
```diff
@@ -65,6 +67,12 @@ public Enrollment signUp(Long eventId) {
         return enrollmentRepository.findByUser(authenticationService.getAuthenticatedUser());
     }
 
+    @Override
+    public byte[] generateQRCode(Long enrollmentId) {
+        return qrcodeService.generateQRCode(
+            String.format("enrollment/%d/confirm", enrollmentId), 640, 640);
```
[the remark this answers is not in the document]
Comment (gajus123) [reply apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/EnrollmentServiceImpl.java]: Done

File: apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/EnrollmentServiceImpl.java [hunk 4]
```diff
@@ -65,6 +69,12 @@ public Enrollment signUp(Long eventId) {
         return enrollmentRepository.findByUser(authenticationService.getAuthenticatedUser());
     }
 
+    @Override
+    public byte[] generateQRCode(Long enrollmentId) {
+        return qrcodeService.generateQRCode(
+            String.format(confirmURLScheme, enrollmentId), 640, 640);
```
Comment (gala377) [inline apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/EnrollmentServiceImpl.java:75]: I would make those magic numbers into `static private final` variables as these are constants

Comment (Delebrith) [reply apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/enrollment/EnrollmentServiceImpl.java:75]: @gajus123 please, make the change and let's merge the PR.

**Pull request merged**